modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainerCluster.java

```diff
@@ -0,0 +1,128 @@
+            .mapToObj(brokerNum ->
+                new KafkaContainer(confluentPlatformVersion)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerNum)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + KafkaContainer.ZOOKEEPER_PORT)
+                    .withEnv("KAFKA_BROKER_ID", brokerNum + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_MIN_ISR", internalTopicsRf + "")
+            )
+            .collect(Collectors.toList());
+    }
+
+    public Network getNetwork() {
+        return this.network;
+    }
+
+    public GenericContainer getZooKeeper() {
+        return this.zookeeper;
+    }
+
+    public Collection<KafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(KafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    private Stream<GenericContainer> allContainers() {
+        Stream<GenericContainer> genericBrokers = this.brokers.stream().map(b -> (GenericContainer) b);
+        Stream<GenericContainer> zookeeper = Stream.of(this.zookeeper);
+        return Stream.concat(genericBrokers, zookeeper);
+    }
+
+    @Override
+    @SneakyThrows
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(b -> (Startable) b);
+        Startables.deepStart(startables).get(60, SECONDS);
+
+        Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> this.brokers.stream().findFirst()
+                .map(this::clusterBrokers)
+                .filter(brokers -> brokers.split(",").length == this.brokersNum)
+                .isPresent());
+    }
+
+    @SneakyThrows
+    private String clusterBrokers(KafkaContainer c) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        dockerClient
+            .execStartCmd(
+                dockerClient.execCreateCmd(c.getContainerId()).withAttachStdout(true)
+                    .withCmd("sh", "-c", "zookeeper-shell zookeeper:" + KafkaContainer.ZOOKEEPER_PORT + " ls /brokers/ids | tail -n 1").exec().getId()
+                )
+            .exec(new ExecStartResultCallback(outputStream, null))
+            .awaitCompletion();
+        return outputStream.toString();
+    }
+
+    @Override
+    public void stop() {
+        // TODO: is there an inverse of `deepStart` to stop all dependant containers?  stopping zookeeper didn't seem to stop brokers alone.
+        allContainers().parallel().forEach(GenericContainer::stop);
+    }
+
+    @Override
+    public void close() {
```

Comment:
not needed, the default method already does that:
https://github.com/testcontainers/testcontainers-java/blob/1d686a1d99050a213572891702e531cacb372ad6/core/src/main/java/org/testcontainers/lifecycle/Startable.java#L18